accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,256 @@
+
+**Q:** Is the tear-off of a `const` constructor a `const` value?
+
+**A:** Yes, but really, all uninstantiated constructor tear-offs are constant values (just like all uninstantiated static/top-level method tear-offs). It doesn't matter whether the constructor is `const` or not. An *instantiated* tear-off (like `List<String> Function(int, String) makeList = List.filled;`) will be constant if the inferred type arguments are constant types, which they are if they contain no type variables (again, just as for static function tear-offs).
+
+**Q:** Can I do a `const` invocation of a torn-off `const` constructor?
+
+**A:** No. When you tear off a constructor, `const` or not, it results in a function value. At that point, all the language knows about it is its function type. You can only invoke constructors with `const` (or `new`), not arbitrary functions, and that value is not a constructor (it's a function which calls a constructor). To create a new constant value, you must specify the constant constructor directly in the `const` constructor invocation.
+
+**Q:** Are tear-offs canonicalized? When are they equal?
+
+**A:** Constants tear-offs are canonicalized. Non-constant tear-offs do try to be equal when they refer to the same constructor, and if type-instantiated, the same constructor with "the same" type arguments. There are complications when going through type aliases, so try to avoid that.
+
+## Named unnamed constructor
+
+A class name, like `DateTime`, evaluates to a `Type` object, which means we have no way to tear off the "unnamed constructor". To allow that, we allow you to refer to the unnamed constructor as `DateTime.new` *as well*.
+
+Everywhere you can currently refer to an unnamed constructor, you will also be able write the same name followed by `.new`. It means exactly the same thing as the unnamed constructor. Everywhere you can use a named constructor, you can use `.new` to refer to the unnamed constructor as if it was named.
+
+Example:
+
+```dart
+class C<T> {
+  const C.new();
+  C.named() : this.new();
+  /// Calls [C.new].
+  factory C.otherNamed() = C<T>.new;  
+}
+class D {
+  D() : super.new();
+}
+void main() {
+  var cs = [C<int>.new(), const C<int>.new(), new C<int>.new()];
+  var tearoff = C.new; // New!
+  var explicitlyTypedTearoff = C<int>.new; // New!
+}
+```
+
+Everywhere except the tear-offs, you can remove the `.new`  and it means the same thing. The tearoff is the only place which *requires* the `.new`.
+
+**Q:** Should I use `.new` or not. What does the style guide say?
+
+**A:** The style guide says nothing yet. The general rule is to not write something which isn't necessary, so don't write `new` unless you need to. For now, only use it for tear-offs and possibly DartDoc (it will be more Dart-like to write `[C.new]` than the current DartDoc-only `[C.C]` to refer to the unnamed constructor). _Don't ever write `new Foo.new()`._
```

Comment:
Personally I'm not a fan of this.  The reason (as I understand it) that we're not making any recommendations in the style guide yet is that we don't want to unduly influence people in one direction or another during this early time period where Dart coders are experimenting and figuring out what kind of style works best.  By making a specific recommendation here, we're subverting that because (hopefully) a lot of people are going to read this FAQ and follow its recommendations, and then when we later look to see what kind of style the community adopted, we'll just see that they adopted our recommendation.

I would rather that we said something like:

> **A:** The style guide says nothing yet.  As with most language features, we are deliberately avoiding making any style recommendations early on so that we can see what kind of a style the community as a whole prefers, and the consenquences of people's stylistic choices.  Once the feature has been in use for a while we'll settle on some specific stylistic recommendations.

Reply:
I agree. I could see that people were trying to introduce lints anyway, so I'd give them this, but it is probably better to not go there. Will remove the recommendation.